src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -292,12 +308,21 @@ var ReactComponentTreeHook = {
   },
 
   getRootIDs() {
-    return Object.keys(rootIDs);
+    var registeredIDs = ReactComponentTreeHook.getRegisteredIDs();
+    return registeredIDs.filter(ReactComponentTreeHook.isRootID);
```

Comment:
This is a rare operation (I currently only use it in tree hook tests) so it’s fine to be slow.